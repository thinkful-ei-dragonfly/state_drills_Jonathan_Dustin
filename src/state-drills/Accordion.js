import React from 'react';

class Accordion extends React.Component{
  static defaultProps = {sections: []}

  state = {
    currentSectionIndex: null,
  }

  handleClick = (index) => {
    this.setState({
      currentSectionIndex: index,
    })
  }

 
  
  renderSections(item, index){
   
      return (
      <li key={index}>
        <button  onClick={() => this.handleClick(index)}>
        {item.title}
        </button>
        {(this.state.currentSectionIndex === index) && <p>{item.content}</p>}
      </li>)
    }
  

  render(){

    return (
    <ul>
      {this.props.sections.map((item,index) => 
        this.renderSections(item, index)
      )}
    </ul>)
  }
}

export default Accordion;