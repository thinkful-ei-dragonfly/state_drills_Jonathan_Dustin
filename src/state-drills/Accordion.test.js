import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe('Accordion Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<Accordion sections={[
        {
          title: 'Buddy',
          content: 'Hello, buddy!',
        },

        {
          title: 'Guy',
          content: `I'm not your buddy, guy!`,
        }
      ]
      } />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
  
  it('renders an empty li when sections prop is not supplied', () => {
    const tree = renderer
      .create(<Accordion />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
  
  it('renders the no sections by default', () => {
    const wrapper = shallow(<Accordion sections={[
      {
        title: 'Buddy',
        content: 'Hello, buddy!',
      },

      {
        title: 'Guy',
        content: `I'm not your buddy, guy!`,
      }
    ]} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens a clicked section', () => {
  const wrapper = shallow(<Accordion sections={[
    {
      title: 'Buddy',
      content: 'Hello, buddy!',
    },

    {
      title: 'Guy',
      content: `I'm not your buddy, guy!`,
    }
  ]} />)
  wrapper.find('button').at(1).simulate('click')
  expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens only last clicked section', () => {
    const wrapper = shallow(<Accordion sections={[
      {
        title: 'Buddy',
        content: 'Hello, buddy!',
      },
  
      {
        title: 'Guy',
        content: `I'm not your buddy, guy!`,
      }
    ]} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
    })

})
