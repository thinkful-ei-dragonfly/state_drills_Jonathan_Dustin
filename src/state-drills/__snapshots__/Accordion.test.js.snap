// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Buddy
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Guy
    </button>
    <p>
      I'm not your buddy, guy!
    </p>
  </li>
</ul>
`;

exports[`Accordion Component opens only last clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Buddy
    </button>
    <p>
      Hello, buddy!
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Guy
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders an empty li when sections prop is not supplied 1`] = `<ul />`;

exports[`Accordion Component renders the UI as expected 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Buddy
    </button>
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Guy
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders the no sections by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Buddy
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Guy
    </button>
  </li>
</ul>
`;
